eclipse-collections-api/src/main/java/org/eclipse/collections/api/set/FixedSizeSet.java

```diff
@@ -19,6 +19,18 @@
 public interface FixedSizeSet<T>
         extends MutableSet<T>, FixedSizeCollection<T>
 {
+    @Override
+    MutableSet<T> with(T element);
+
+    @Override
+    MutableSet<T> without(T element);
+
+    @Override
+    MutableSet<T> withAll(Iterable<? extends T> elements);
+
+    @Override
+    MutableSet<T> withoutAll(Iterable<? extends T> elements);
```

Comment:
I'm wondering, can without and withoutAll return FixedSize*?

Reply:
Nope, same as the others. for example `AbstractMemoryEfficientMutableList` that implements `FixedSizeList` return MutableList
```
    @Override
    public MutableList<T> withoutAll(Iterable<? extends T> elements)
    {
        if (Iterate.isEmpty(elements))
        {
            return this;
        }
        return Lists.fixedSize.ofAll(this.toList().withoutAll(elements));
    }
```

Reply:
???

This compiles fine for me...

```java
@Override
public FixedSizeList<T> withoutAll(Iterable<? extends T> elements)
{
    if (Iterate.isEmpty(elements))
    {
        return this;
    }
    return Lists.fixedSize.ofAll(this.toList().withoutAll(elements));
}
```

Reply:
Anyway, this change can't be made in a minor version, so I think we're ready to land this change as-is.

Reply:
Yes, in that cyaee, the same changes needs to be made in every class that implements `FixedSize*`! Is this something of interest? I can introduce all the discussions captured here in another PR

Reply:
@aboullaite Don't make those changes yet.  Leave them as is for now.  We can fix this during a major release.

Reply:
@aboullaite Yes you can do this in a separate PR, but it will have to wait for a major release.